src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java

```diff
@@ -25,29 +28,50 @@
 import uk.gov.hmcts.reform.sscs.model.single.hearing.MemberType;
 import uk.gov.hmcts.reform.sscs.model.single.hearing.PanelPreference;
 import uk.gov.hmcts.reform.sscs.model.single.hearing.PanelRequirements;
+import uk.gov.hmcts.reform.sscs.reference.data.model.PanelCategory;
 import uk.gov.hmcts.reform.sscs.reference.data.model.SessionCategoryMap;
+import uk.gov.hmcts.reform.sscs.reference.data.service.PanelCategoryService;
 import uk.gov.hmcts.reform.sscs.service.holder.ReferenceDataServiceHolder;
 
 @Slf4j
+@Component
 public final class HearingsPanelMapping {
 
-    private HearingsPanelMapping() {
+    private final PanelCategoryService panelCategoryService;
 
+    @Value("${feature.default-panel-comp.enabled}")
+    private boolean defaultPanelCompEnabled;
+
+    HearingsPanelMapping(PanelCategoryService panelCategoryService) {
+
+        this.panelCategoryService = panelCategoryService;
     }
 
-    public static PanelRequirements getPanelRequirements(SscsCaseData caseData,
+    public PanelRequirements getPanelRequirements(SscsCaseData caseData,
                                                          ReferenceDataServiceHolder refData) {
         return PanelRequirements.builder()
-                .roleTypes(getRoleTypes(caseData.getBenefitCode()))
+                .roleTypes(getRoleTypes(caseData))
                 .authorisationTypes(getAuthorisationTypes())
                 .authorisationSubTypes(getAuthorisationSubTypes())
                 .panelPreferences(getPanelPreferences(caseData))
                 .panelSpecialisms(getPanelSpecialisms(caseData, getSessionCaseCodeMap(caseData, refData)))
                 .build();
     }
 
-    public static List<String> getRoleTypes(String benefitCode) {
-        return BenefitRoleRelationType.findRoleTypesByBenefitCode(benefitCode);
+    public List<String> getRoleTypes(SscsCaseData caseData) {
```

Comment:
would it be possible to move this method to common because we'll probably need it for the migration